### PR TITLE
Democracy: Improve tests

### LINKED ISSUE #91: Improve democracy template tests
Current democracy template tests can only be run with geth. We should provide support for ganache as well. Additionally, there are multiple steps along the suite that can be simplified using some of the helpers provided in `test-helpers` and `templates-shared`.

### PR DESCRIPTION
**Pull request by facuspagnuolo**
**CHANGE BASE BRANCH TO MASTER AFTER #93 GETS MERGED**

Fixes #91

### CODE REVIEW

File: templates/democracy/test/gas.js [hunk 1]
```diff
@@ -1,37 +1,36 @@
...
+    it('create token', async () => {
+      await template.newToken('DemocracyToken', 'DTT', { from: owner })
+    })
+
+    it('create new instance', async () => {
+      const holders = [holder20, holder29, holder51]
+      const stakes = [20e18, 29e18, 51e18]
+
+      const id = 'DemocracyDao-' + Math.random() * 1000
+      await template.newInstance(id, holders, stakes, neededSupport, minimumAcceptanceQuorum, votingTime, { from: owner })
```
Comment (sohkai) [inline templates/democracy/test/gas.js]: I remember fixing something similar in these templates before, but we should be careful whenever we use this randomization to floor or ceil it. Right now, it still results in decimals, that will create an ENS subdomain.

File: templates/democracy/test/democracy.js [hunk 2]
```diff
@@ -1,402 +1,316 @@
...
+  const STAKES = [20e18, 29e18, 51e18]
+  const HOLDERS = [holder20, holder29, holder51]
+
+  const increaseTime = async s => {
+    if (isGanache()) return timeTravel(s)
+    const previousTime = await (await Timestamp.new()).getNow()
+    await new Promise(resolve => setTimeout(resolve, s * 1000))
+    const currentTime = await (await Timestamp.new()).getNow()
+    assert.isAtLeast(currentTime.minus(s).toNumber(), previousTime.toNumber(), `sleep/time-travel helper failed to increase ${s} seconds`)
+  }
```
Comment (sohkai) [inline templates/democracy/test/democracy.js]: Should we pull this off into a shared helper?

File: templates/democracy/test/democracy.js [hunk 3]
```diff
@@ -1,402 +1,316 @@
...
-    const sleep = function(s) {
-        return new Promise(resolve => setTimeout(resolve, s*1000));
-    }
+        context('> Finance access', () => {
+          let voteId
+          const payment = new web3.BigNumber(2e16)
+
+          beforeEach('fund finance', async () => {
+            await finance.sendTransaction({ value: payment, from: owner })
+            const paymentAction = { to: finance.address, calldata: finance.contract.newImmediatePayment.getData(ETH, nonHolder, payment, 'voting payment') }
```
Comment (sohkai) [inline templates/democracy/test/democracy.js]: Bit of a nitpick, technically this isn't really "funding" finance and could be moved to its own `beforeEach()`

File: templates/democracy/test/democracy.js (hunk 2 above)
Comment (facuspagnuolo) [reply templates/democracy/test/democracy.js]: sure! will do that 👍

**Pull request merged**